coap-core/src/main/java/com/mbed/coap/transport/udp/DatagramSocketTransport.java

```diff
@@ -134,39 +139,65 @@ protected void receive(CoapReceiver coapReceiver, DatagramPacket datagramPacket)
         }
     }
 
-    protected DatagramSocket createSocket() throws SocketException {
-        DatagramSocket datagramSocket = new DatagramSocket(bindSocket);
-        datagramSocket.setSoTimeout(200);
-        return datagramSocket;
-    }
-
     @Override
     public void stop() {
-        if (socket != null) {
-            socket.close();
-        }
+        closeSocket();
         TransportExecutors.shutdown(readingWorker);
     }
 
     @Override
     public void sendPacket0(CoapPacket coapPacket, InetSocketAddress adr, TransportContext transContext) throws CoapException, IOException {
-        if (socket == null) {
+        if (!socketCreated()) {
             throw new IllegalStateException();
         }
         byte[] data = coapPacket.toByteArray();
 
         DatagramPacket datagramPacket = new DatagramPacket(data, data.length, adr);
 
-        socket.send(datagramPacket);
+        sendPacket(datagramPacket);
+    }
+
+    protected boolean socketCreated() {
+        return (socket != null);
+    }
+
+    protected void updateReceiveBufferSize() throws SocketException {
+        this.socket.setReceiveBufferSize(socketBufferSize);
+    }
+
+    protected void updateSendBufferSize() throws SocketException {
+        this.socket.setSendBufferSize(socketBufferSize);
+
+    }
+
+    protected void updateReuseAddress() throws SocketException {
+        this.socket.setReuseAddress(reuseAddress);
+    }
+
+    protected void receivePacket(DatagramPacket datagramPacket) throws IOException {
+        this.socket.receive(datagramPacket);
+
+    }
+
+    protected void sendPacket(DatagramPacket packet) throws IOException {
+        this.socket.send(packet);
+    }
+
+    protected void createSocket() throws SocketException {
+        socket = new RealDatagramSocket(bindAddress);
+        socket.setSoTimeout(200);
     }
```

Comment:
Looks like those new methods are not used anywhere.

Reply:
fixed.